### ISSUE TITLE
Tags unmatched by any components create vars of the last traversed element's type

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular_analyzer_plugin#169]
I get the error for `!isSupported(prop)` for

```html
  <template ngFor let-prop [ngForOf]="individualProperties">
    <div>
      <template [ngIf]="!isSupported(prop) && prop.typeName != 'Reference'">
        <div>resource-detail-component (unsupported {{prop.typeName}})</div>
      </template>
    </div>
  </template>
```

```dart
bool isSupported(PropertyMetadata prop) => ...
```

**Comment by zoechi** [comment]
Seems to be caused by the explicit template element form 

```html
<template ngFor let-prop [ngForOf]="individualProperties">
```

I have other cases as well where I have this form.

**Comment by MichaelRFairhurst** [comment]
Do you use `#prop` on a div anywhere in this template?

**Comment by zoechi** [comment]
No, only `let-prop`

I also have `<th *ngFor="let prop of summaryProperties">{{prop.name}}</th>` in the same template, but they are siblings, not nested.

**Comment by MichaelRFairhurst** [comment]
The logic for honoring template scopes is pretty hairy right now. I could see this being a data race that confuses the div for the template. I have a branch that rewrites that, hopefully resulting in fewer of these strange bugs rather than more of them!

@mk13 could you try to reproduce this? Ideally with a failing test in resolver_test.dart? I want to see it failing without my refactor before we say with confidence that the refactor fixes it.

**Comment by zoechi** [comment]
I just see these errors are gone now as well.

**Comment by MichaelRFairhurst** [comment]
I'm still worried that it was caused by a data race, perhaps the conditions have changed to where it is no longer rearing its head.

But it might be very hard to hunt down.

**Comment by zoechi** [comment]
This seems to work for me now

**Comment by MichaelRFairhurst** [comment]
Ah yes, I believe Max fixed this along with #348
